### PR TITLE
fix: update 0069-VCBS-058 and 0069-VCBS-054

### PR DESCRIPTION
**Pull request by claudiumilea**

### CODE REVIEW

File: protocol/0069-VCBS-validators_chosen_by_stake.md [hunk 1]
```diff
@@ -13,12 +13,14 @@ At the end of each epoch Vega will calculate the unnormalised `validator_score`,
...
+If `empty_slots := network.validators.tendermint.number - n > 0` (we have empty consensus (Tendermint) validator slots) then the top `empty_slots` from `[w_1, ..., w_k]` are promoted to consensus (Tendermint) validators. 
+If `w_1>v_n` (i.e. the highest scored potential validator has more than the lowest score incumbent validator) then in the new epoch `w_1` becomes a consensus forming (Tendermint) validator, and the lowest scoring incumbent becomes an ersatz validator. 
+The exception to that rule is if one or more incumbent validators drop below the required ownstake (ownstake < reward.staking.delegation.minimumValidatorStake), either through changing their self-delegation or due to a change of the network parameter. 
+In that case, the validator with the smallest ownstake (which is smaller than reward.staking.delegation.minimumValidatorStake) is demoted, and the validator score is not used for demotion in that epoch.
```
Comment (ze97286) [inline protocol/0069-VCBS-validators_chosen_by_stake.md]: I don't understand the wording of this - if the validator has less than the required stake they will have a ranking score of 0 so if there is a better ersatz validator they will be swapped in/out. I don't understand what you mean by "and the validator score is not used for demotion in that epoch."

File: protocol/0069-VCBS-validators_chosen_by_stake.md [hunk 2]
```diff
@@ -300,6 +302,46 @@ See [limited network life spec](./0073-LIMN-limited_network_life.md).
...
+  * Run with `network.validators.ersatz.multipleOfTendermintValidators = 1`
+  * Setup a network with 6 consensus forming (Tendermint) validators
+  * Ensure that the multisig is updated to those 6 validators.
+  * Ensure that the threshold on the multisig is set to `666`.
+  * Change the network parameter `network.validators.tendermint.number` to 3 Tendermint validators.
+  * Verify that exactly one consensus forming validator with the lowest score is demoted to an ersatz validator at the beginning of the next epoch and we are running with 5 consensus (Tendermint) validators. 
+  * Ensure that the multisig is updated to those 5 validators.
+  * Verify that exactly one consensus forming validator with the lowest score is demoted to an ersatz validator at the beginning of the following epoch and we are running with 4 consensus (Tendermint) validators.
+  * Ensure that the multisig is *not* updated to those 4 validators, but we have the 5 validators from previous step.
+  * Verify that no consensus forming validator is removed at the start of the next epoch and we are running with 3 consensus (Tendermint) validators.
```
Comment (ze97286) [inline protocol/0069-VCBS-validators_chosen_by_stake.md]: this doesn't seem right or is just unclear? in line 340 we're left with 5 validators, in line 341 we're left with demoting another one, left with 4 validators running consensus but 5 on the bridge, but then line 343 says: `Verify that no consensus forming validator is removed ` however `and we are running with 3 consensus (Tendermint) validators.` - so I don't understand how the two coexist?

File: protocol/0069-VCBS-validators_chosen_by_stake.md (hunk 1 above)
Comment (davidsiska-vega) [reply protocol/0069-VCBS-validators_chosen_by_stake.md]: In my mind it was just: if you don't have enough `ownStake` then it doesn't matter what your score is, you'll be out. This probably comes from insufficiently tight definition of "validator score".

Comment (davidsiska-vega) [inline protocol/0069-VCBS-validators_chosen_by_stake.md:23]:
```suggestion
In that case, the validator with the smallest ownstake (which is smaller than reward.staking.delegation.minimumValidatorStake) is demoted.
```

File: protocol/0069-VCBS-validators_chosen_by_stake.md [hunk 3]
```diff
@@ -300,6 +302,46 @@ See [limited network life spec](./0073-LIMN-limited_network_life.md).
...
+3.e Demote a number of consensus forming (Tendermint) validators due to lack of slots (<a name="0069-VCBS-065" href="#0069-VCBS-065">0069-VCBS-065</a>):
+  * Run with `network.validators.ersatz.multipleOfTendermintValidators = 1`
+  * Setup a network with 6 consensus forming (Tendermint) validators
+  * Ensure that the multisig is updated to those 6 validators.
+  * Ensure that the threshold on the multisig is set to `666`.
+  * Change the network parameter `network.validators.tendermint.number` to 3 Tendermint validators.
+  * Verify that exactly one consensus forming validator with the lowest score is demoted to an ersatz validator at the beginning of the next epoch and we are running with 5 consensus (Tendermint) validators. 
+  * Ensure that the multisig is updated to those 5 validators.
+  * Verify that exactly one consensus forming validator with the lowest score is demoted to an ersatz validator at the beginning of the following epoch and we are running with 4 consensus (Tendermint) validators.
+  * Ensure that the multisig is *not* updated to those 4 validators, but we have the 5 validators from previous step.
```
Comment (davidsiska-vega) [inline protocol/0069-VCBS-validators_chosen_by_stake.md:342]:
```suggestion
  * Ensure that the multisig is updated to those 4 validators. 
```

File: protocol/0069-VCBS-validators_chosen_by_stake.md (hunk 1 above)
Comment (ze97286) [inline protocol/0069-VCBS-validators_chosen_by_stake.md]: that's not how it's implemented. either you have sufficient stake or you don't - if you don't, it doesn't matter if you have a fraction less or you have 0, your ranking score is 0. If you have two validators with equal ranking score, the one that's with us longer will be first in the sorting order (so will be swapped last).

File: protocol/0069-VCBS-validators_chosen_by_stake.md (hunk 2 above)
Comment (davidsiska-vega) [inline protocol/0069-VCBS-validators_chosen_by_stake.md]:
```suggestion
  * Verify that no consensus forming validator is removed at the start of the next epoch and we are running with 4 consensus (Tendermint) validators. 
```

File: protocol/0069-VCBS-validators_chosen_by_stake.md (hunk 1 above)
Comment (davidsiska-vega) [inline protocol/0069-VCBS-validators_chosen_by_stake.md]:
```suggestion
In that case, the validators that have less than the required own stake get a ranking score is 0. If you have two validators with equal ranking score, the one that's been a consensus validator longer will be first in the sorting order (so will be swapped last).
```

File: protocol/0069-VCBS-validators_chosen_by_stake.md (hunk 2 above)
Comment (davidsiska-vega) [reply protocol/0069-VCBS-validators_chosen_by_stake.md]: there was a typo in the numbers - fixed - is it ok now?

Comment (ze97286) [reply protocol/0069-VCBS-validators_chosen_by_stake.md]: yes

**Review by davidsiska-vega — APPROVED** [review]

**Pull request merged**